src/addDebouncedHandler.tsx

```diff
@@ -0,0 +1,42 @@
+import {addHandlers, UnchangedProps, flowMax} from 'ad-hok'
+import {debounce} from 'lodash'
+
+import addPropTrackingRef from './addPropTrackingRef'
+import cleanupProps from './cleanupProps'
+
+type AddDebouncedHandlerType = <
+  TPropName extends string,
+  TProps extends {[handlerName in TPropName]: (...args: any[]) => any}
+>(
+  waitInterval: number,
+  handlerPropName: TPropName,
+) => UnchangedProps<TProps>
+
+const addDebouncedHandler: AddDebouncedHandlerType = (
+  waitInterval,
+  handlerPropName,
+) => {
+  const refName = 'addDebouncedHandlerRef'
+  const invokeRefHandlerName = 'addDebouncedHandlerInvokeRef'
+
+  return flowMax(
+    addPropTrackingRef(handlerPropName, refName),
+    addHandlers(
```

Comment:
The existing (ie from the current project) implementation of `addDebouncedHandler()` exposed a dependencies argument, which is a confusing concept when it comes to debouncing a handler/its identity

So updated so that (by using a ref internally) it should always preserve stable identity of the handler being debounced and the exposed debounced handler

I guess that means that you couldn't force resetting the debounce interval by having the handler identity change, but that seems like an unusual use case?